torch/_inductor/codegen/cpp.py

```diff
@@ -1994,10 +2011,35 @@ def __init__(self, kernel_group):
 
     def legalize_bf16(self, nodes):
         def add_to_dtype(sub_graph: torch.fx.Graph):
+            def is_bf16_mem_copy(node: torch.fx.Node):
+                if node.target in ["load", "constant"]:
+                    bf16_mem_copy = all(
+                        usr.target == "store"
+                        and V.graph.get_dtype(usr.args[1]) == torch.bfloat16
+                        for usr in node.users
+                    )
+                    if bf16_mem_copy:
+                        opt_ctx = OptimizationContext()
+                        opt_ctx.is_bf16_mem_copy = bf16_mem_copy
+                        node.meta[OptimizationContext.key] = opt_ctx
+                    return bf16_mem_copy
+                elif node.target is ["store"]:
+                    stored_node = node.args[3]
+                    bf16_mem_copy = is_bf16_mem_copy(stored_node)
+                else:
```

Comment:
The target of a node could be `get_index` and the semantic is still `memcpy`. Please check it.

Reply:
I think we do not need care about `get_index`.
 - we only invoke `is_bf16_mem_copy` to check whether need to insert `to_dtype` when `node.target in ("load", "store", "constant")`.
 - Maybe we should assert it in `is_bf16_mem_copy`?